### PR TITLE
Add model export utilities for YOLOv8 and YOLOv9

### PR DESCRIPTION
**Pull request by spal-synaptics**
- Added export script to export YOLO model to tflite / onnx / torch / tensorflow + generate YAML metafile for onnx and tflite
- Added convert script to convert exported model to SyNAP
  - Currently using the SyNAP docker, will upgrade to the pip library once #3 is merged
- Added scripts to copy models to board and profile with `synap_cli`
  - Currently via SSH, will add ADB support in the future

See the quickstart in /model/README.md for instructions on using these scripts.

### CODE REVIEW

File: model/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+This toolkit provides a convenient way of exporting models to the SyNAP format.
+The following models are currently supported:
+
+| Model | Export Script |
+|-------|---------------|
+| [YOLOv8](https://docs.ultralytics.com/models/yolov8/) | [export/yolo.py](/model/export/yolo.py) |
+| [YOLOv9](https://docs.ultralytics.com/models/yolov9/) | [export/yolo.py](/model/export/yolo.py) |
+
+## YOLO Export Quickstart
+This tutorial will go through exporting the YOLOv8 segmentation (small) model into two input sizes (640x353, 224x224) and two quantization types (uint8, float16). The quantization dataset used is [`coco8-seg`](https://docs.ultralytics.com/datasets/segment/coco8-seg/) and is assumed to be located at `"/home/$USER/synaptics-synap/toolkit/datasets/coco8-seg/*.jpg"`. The board used is the Astra SL1680 with firmware v1.1.0.
```
Comment (Meet-synaptics) [inline model/README.md]: typo 640*353 => 640*352

File: model/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# SyNAP Model Export
+This toolkit provides a convenient way of exporting models to the SyNAP format.
+The following models are currently supported:
+
+| Model | Export Script |
+|-------|---------------|
+| [YOLOv8](https://docs.ultralytics.com/models/yolov8/) | [export/yolo.py](/model/export/yolo.py) |
```
Comment (Meet-synaptics) [inline model/README.md:7]: What are the limitations here to generalize this tool for any version of YOLO model?

File: model/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+| [YOLOv9](https://docs.ultralytics.com/models/yolov9/) | [export/yolo.py](/model/export/yolo.py) |
+
+## YOLO Export Quickstart
+This tutorial will go through exporting the YOLOv8 segmentation (small) model into two input sizes (640x353, 224x224) and two quantization types (uint8, float16). The quantization dataset used is [`coco8-seg`](https://docs.ultralytics.com/datasets/segment/coco8-seg/) and is assumed to be located at `"/home/$USER/synaptics-synap/toolkit/datasets/coco8-seg/*.jpg"`. The board used is the Astra SL1680 with firmware v1.1.0.
+
+### 1. Export YOLOv8 to TFLite models
+```
+python -m model.export.yolo \
+    --models yolov8s-seg \
```
Comment (Meet-synaptics) [inline model/README.md:16]: can it export classification/object-detection/pose models?

**Comment by Meet-synaptics** [comment]
We could also rename 'model' to something 'export_scripts' which might be a more descriptive name

File: model/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,81 @@
+absl-py==2.1.0
```
Comment (Meet-synaptics) [inline model/requirements.txt]: We also need to exclude this all requirements.txt files while building the pip package. The current implementation of the wheel build includes all requirements.txt files within the toolkit repository. I think we can avoid including these requirements in the Python wheel?

File: model/README.md (hunk 2 above)
Comment (spal-synaptics) [reply model/README.md:7]: The current limitation is that the model export has to happen through the ultralytics package, specifically through `Model.export()`. So we can't do, for example, YOLOv5 as that's it's own repository and not through the ultralytics package.
That being said, the model exporter is modular. So someone could just subclass the base `ModelExporter` and provide their own `export_model()`  for any model. All they need to ensure is that `export_model()` should return the path to the exported tflite/onnx/torch etc. model.

File: model/README.md (hunk 3 above)
Comment (spal-synaptics) [reply model/README.md:16]: Yep, should be able to export all models supported by ultralytics's `Model.export()`.

**Review by Meet-synaptics — APPROVED** [review]

**Review by mdenna-synaptics — APPROVED** [review]
looks good to me, the only comment as already suggested by Meet is to rename "model" folder to something more descriptive of the content, such as "export".

**Comment by spal-synaptics** [comment]
> looks good to me, the only comment as already suggested by Meet is to rename "model" folder to something more descriptive of the content, such as "export".

@mdenna-synaptics @Meet-synaptics How about "model_tools" as a name? I can't use "export" as that's already a sub-module and the module does do non-export tasks like copying models to the board and profiling them.

**Comment by mdenna-synaptics** [comment]
> > looks good to me, the only comment as already suggested by Meet is to rename "model" folder to something more descriptive of the content, such as "export".
> 
> @mdenna-synaptics @Meet-synaptics How about "model_tools" as a name? I can't use "export" as that's already a sub-module and the module does do non-export tasks like copying models to the board and profiling them.

I see.. or maybe just "tools" or "utils" would also look good to me..

**Pull request merged**